presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveBasicTableStatistics.java

```diff
@@ -0,0 +1,400 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests.hive;
+
+import com.google.common.primitives.Longs;
+import io.prestodb.tempto.ProductTest;
+import io.prestodb.tempto.Requires;
+import io.prestodb.tempto.fulfillment.table.hive.tpch.ImmutableTpchTablesRequirements.ImmutableNationTable;
+import io.prestodb.tempto.query.QueryExecutor;
+import io.prestodb.tempto.query.QueryResult;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.OptionalLong;
+
+import static com.facebook.presto.tests.TestGroups.HIVE_CONNECTOR;
+import static com.facebook.presto.tests.TestGroups.HIVE_TABLE_STATISTICS;
+import static com.facebook.presto.tests.utils.QueryExecutors.onHive;
+import static com.facebook.presto.tests.utils.QueryExecutors.onPresto;
+import static com.google.common.base.Verify.verify;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@Requires(ImmutableNationTable.class)
+public class TestHiveBasicTableStatistics
```

Comment:
can you add tests with EXTERNAL non empty tables without stats, partitioned and non partitioned. What is happening when you do an INSERT to such tables?

Reply:
If writes to non managed tables is disabled (`hive.non-managed-table-writes-enabled`) exception is raised. If writes are enabled it should do the same way as for managed tables. 

This PR does not cover the case when statistics recomputing is needed. Like when you are inserting into an existing table with no statistics computed for the data it already has. For such cases the real rows count, and rows count in statistics will be different.

Reply:
> If we are inserting into existing table/partition, and statistics are not present - than throw an exception

you don't need to throw in that case. You can simply skip saving stats.

Reply:
Yeah, that's a good idea. I haven't thought about that. Will do.

Reply:
@findepi done

Reply:
>> If we are inserting into existing table/partition, and statistics are not present - than throw an exception

>you don't need to throw in that case. You can simply skip saving stats.

Do you have tests for this case?

Reply:
I added unit tests. Can add product tests as well.

Reply:
UPD.

Turns out Hive always sets statistics, even on `CREATE EXTERNAL TABLE`. Those are just zero if the table is empty. 

Also Hive doesn't allow you to overwrite table statistics parameters by doing `ALTER TABLE nation SET TBLPROPERTIES ('totalSize' = 1234);`

The only way to reset those statistics is to directly connect to the metastore and override those parameters. Since connecting to metastore and changing something directly is not quite a "product test" i just decided to implement an integration test.